src/platform/packages/shared/response-ops/rule_params/custom_threshold/v1.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { schema } from '@kbn/config-schema';
+import { COMPARATORS } from '@kbn/alerting-comparators';
+import { dataViewSpecSchema } from '../common';
+import { oneOfLiterals, validateKQLStringFilter, LEGACY_COMPARATORS } from '../common/utils';
+
+enum Aggregators {
+  COUNT = 'count',
+  AVERAGE = 'avg',
+  SUM = 'sum',
+  MIN = 'min',
+  MAX = 'max',
+  CARDINALITY = 'cardinality',
+  RATE = 'rate',
+  P95 = 'p95',
+  P99 = 'p99',
+  LAST_VALUE = 'last_value',
+}
+
+const comparators = Object.values({ ...COMPARATORS, ...LEGACY_COMPARATORS });
+
+const allowedAggregators = Object.values(Aggregators);
+allowedAggregators.splice(Object.values(Aggregators).indexOf(Aggregators.COUNT), 1);
+
+export const searchConfigurationSchema = schema.object({
+  index: schema.oneOf([schema.string(), dataViewSpecSchema]),
+  query: schema.object({
+    language: schema.string(),
+    query: schema.string({
+      validate: validateKQLStringFilter,
+    }),
+  }),
+  filter: schema.maybe(
+    schema.arrayOf(
+      schema.object({
+        query: schema.maybe(schema.recordOf(schema.string(), schema.any())),
+        meta: schema.recordOf(schema.string(), schema.any()),
+      })
+    )
+  ),
+});
+
+const baseCriterion = {
+  threshold: schema.arrayOf(schema.number()),
+  comparator: oneOfLiterals(comparators),
+  timeUnit: schema.string(),
+  timeSize: schema.number(),
+};
+
+const customCriterion = schema.object({
+  ...baseCriterion,
```

Comment:
nit: if `baseCriterion` is only used here I don't know if it makes sense to be a different variable.